mozilla_schema_generator/main_ping.py

```diff
@@ -31,17 +31,46 @@ def get_schema(self):
         return self._update_env(schema)
 
     def _update_env(self, schema):
-        user_pref_elem = {
+        integer = {"type": "integer"}
+        string = {"type": "string"}
+        string_map = {
+            "type": "object",
+            "additionalProperties": string,
+        }
+        nested_string_map = {
+            "type": "object",
+            "additionalProperties": string_map,
+        }
+        map_without_values = {
             "type": "object",
             "additionalProperties": {
-                "type": "string"
-            }
+                "type": "object",
+            },
         }
 
+        active_addons = prepend_properties(("environment", "addons", "activeAddons")) + ("additionalProperties", "properties")
+        simple_measurements = prepend_properties(("payload", "simpleMeasurements", ""))[:-1]
+        processes_dynamic = prepend_properties(("payload","processes", "dynamic", ""))[:-1]
+
+        schema.set_schema_elem(
+                prepend_properties(("environment", "settings", "userPrefs")), string_map)
+        schema.set_schema_elem(
+                prepend_properties(("environment", "system", "os", "version")), string)
```

Comment:
This should already be a string [in the schema](https://github.com/mozilla-services/mozilla-pipeline-schemas/blob/2babf5c3c116076e663f96a8e3f1eaf710063868/schemas/telemetry/main/main.4.schema.json#L1060-L1064). See also https://github.com/mozilla-services/mozilla-pipeline-schemas/pull/420

Reply:
It looks like this is from existing code, but removing this should not affect the resulting schema diff.

Reply:
I would rather remove the existing code as a separate PR